cave/utils/csv2rh.py

```diff
@@ -0,0 +1,250 @@
+                config_data = config_data.apply(pd.to_numeric, errors='ignore')
+                parameters = ['p_' + p for p in config_data.columns]
+                # Create and fill in p_-columns for cs-creation
+                for p in parameters:
+                    self.data[p] = 0
+                def fill_parameters(row):
+                    for p in parameters:
+                        row[p] = config_data.loc[row['config_id'], p[2:]]
+                    return row
+                self.data = self.data.apply(fill_parameters, axis=1)
+                self.get_cs()
+                self.id_to_config = {}
+                for index, row in config_data.iterrows():
+                    self.id_to_config[index] = Configuration(self.cs,
+                            values={name[2:] : value for name, value in
+                                             zip(parameters, row)})
+                self.config_to_id = {conf : name for name, conf in
+                        self.id_to_config.items()}
+            elif isinstance(self.id_to_config, dict):
+                self.config_to_id = {conf : name for name, conf in
+                        self.id_to_config.items()}
+                self.get_cs()
+        elif parameters:
+            # Add new column for config-ids
+            self.data['config_id'] = -1
+            # Configs are defined via individual "p_"-header-columns
+            self.get_cs()
+            def add_config(row):
+                config = Configuration(self.cs, values={p[2:] : float(row[p]) for p in
+                                                   parameters})
+                if not config in self.config_to_id.keys():
+                    # New index (unseen config)
+                    new_id = len(self.config_to_id)
+                    self.config_to_id[config] = new_id
+                    self.id_to_config[new_id] = config
+                row['config_id'] = self.config_to_id[config]
+                return row
+            self.data = self.data.apply(add_config, axis=1)
+        else:
+            raise ValueError("Define configs either via \"p_\" or \"config_id\" in header.")
+
+    def _complete_instances(self):
+        self.id_to_inst_feats = {}
+        self.inst_feats_to_id = {}
+        inst_feats = [i for i in self.data.columns if i.startswith('i_')]
+        if inst_feats and 'instance_id' in self.data.columns:
+            raise ValueError("Define instances either via \"i_\" or \"instance_id\" in header, not both.")
+        elif 'instance_id' in self.data.columns:
+            if self.instance_features:
+                inst_feats = self.instance_features[0]
+                self.id_to_inst_feats = {i : tuple(feat) for i, feat in self.instance_features[1]}
+                self.inst_feats_to_id = {feat : i for i, feat in self.id_to_inst_feats}
+            else:
+                self.logger.warning("Instances defined but no instance features available.")
```

Comment:
Is a warning enough here?
Shouldn't we stop with an error?
I'd argue for exiting with an error as the user specified instances but no instance features are available. This makes a lot of the analysis not functioning in cave.

Reply:
Ok. Might aswell.